lib/Dialect/OM/Transforms/LinkModules.cpp

```diff
@@ -84,6 +97,21 @@ void ModuleInfo::postProcess(const SymMappingTy &symMapping) {
       // ClassExternFieldOp will be deleted as well.
       return WalkResult::skip();
     }
+    if (isa<HWModuleExternOp>(op)) {
+      op->erase();
+      return WalkResult::skip();
+    }
+    if (auto hwModule = dyn_cast<HWModuleOp>(op)) {
+      // Update its class name if changed.
+      auto it = symMapping.find({module, hwModule.getNameAttr()});
+      if (it != symMapping.end()) {
+        hwModule.setSymNameAttr(it->second);
+      } else {
+        modReplacer.recursivelyReplaceElementsIn(op, true, false, false);
```

Comment:
does this walks entire operations in the module?

Reply:
Added tests to ensure nested regions are updated.